### ISSUE TITLE
Mixing LineSeries based on dateAxis and ColumnSeries

### ISSUE THREAD

**Issue by ptorrent** [amcharts/amcharts4#130]
Hello,

I'm trying to mix line series based on dateAxis and column series based on categoryAxis.

https://codepen.io/ptorrent/pen/BVXWXE

There's an extra category on the right of the chart  and I can't explain that ?

![image](https://user-images.githubusercontent.com/6083644/42412863-f242f2a2-8214-11e8-877f-08ef3e3b8ed7.png)

If I remove 

`
{
    "date" : new Date(2018,0,1,0,0,0,0),
    value2 : 22
  },{
    "date" : new Date(2018,5,1,0,0,0,0),
    value2 : 29
  },{
    "date" : new Date(2019,0,1,0,0,0,0),
    value2 : 23
  }
`
The display looks good. 

Thanks for your help !

**Comment by martynasma** [comment]
Well, the column series (or more like its axis is thrown off) by additional data points in the data. If you had a data of more or less the same type, e.g. two `DateAxis` or two `CategoryAxis`, it would be fine.

That said, there's definitely something funky going on and we're going to look into it.

Meanwhile, for a quick and easy fix, simply move the LineSeries portion of data into its own `data` property:

```
"series": [ {
  // ...
}, {
  "stroke": "red",
  // ...
  "data": [ {
    "date": new Date( 2018, 0, 1, 0, 0, 0, 0 ),
    value2: 22
  }, {
    "date": new Date( 2018, 5, 1, 0, 0, 0, 0 ),
    value2: 29
  }, {
    "date": new Date( 2019, 0, 1, 0, 0, 0, 0 ),
    value2: 23
  } ]
}]
```

Here's a working pen:
https://codepen.io/team/amcharts/pen/f09879699f5a093f7677b0043560ba8e/

**Comment by ptorrent** [comment]
Thank you very much for your prompt and accurate response.

It would be easier for me to associate the values directly on the ColumSeries, I tried but nothing is displayed

https://codepen.io/ptorrent/pen/qKeKvQ

**Comment by martynasma** [comment]
I know. However, CategoryAxis is also a user of categories. Axis can't use the data stored in Series, hence you need to have it in chart's `data`.

It does not have to be full data, just categories.

Does that make sense?

**Comment by ptorrent** [comment]
If I understand well, there can only be one category axis per chart?

**Comment by martynasma** [comment]
Nope. You can have as many category axes per chart as you wish. However, contrary to other axis types, Category axis needs to know what the actual categories are, and it relies on chart's data to do that.

This is why you need to set `dataFields.category` for the Category axis.

Technically, you should be able to set `data` directly on Category axis as well, but, frankly, I haven't tried that :)

**Comment by ptorrent** [comment]
Thanks,

I tried to create a chart with 2 category axis, if the number of category does not match, the rendering does not work well

https://codepen.io/ptorrent/pen/YvmjbG

**Comment by martynasma** [comment]
Indeed. Something's off.

However, I'm not entirely sure how chart should behave in such case. The data for both categories is located in a single array. Theoretically, I can imagine fuzzy logic that could untangle this, but I'm not sure if it's worth adding all this overhead to core functionality.

That said, adding individual data directly to series AND axes, works exactly as expected:

https://codepen.io/team/amcharts/pen/qKeMZZ

**Comment by ptorrent** [comment]
Thanks for your help and your work.

Version 4 is very promising !

**Comment by ptorrent** [comment]
I'm testing scrollbarX with data update on serie : 

https://codepen.io/ptorrent/pen/KeObwr

The scrollbarX didn't change after data serie update

I didn't find a function to update the scrollbarX in the API after data chage

**Comment by martynasma** [comment]
Regarding 1) there seems to be a bug which prevents series to be drawn correctly when data is set directly on series.

I filled out a separate issue for that: #134.

**Comment by martynasma** [comment]
Regarding cursor. Yes, it seems cursor is failing if its chart does not have any data at all.

Moving the data from scrollbar to chart itself, should fix the issue.

**Comment by ptorrent** [comment]
Thanks for your support !

You're right (for cursor)  : 

https://codepen.io/ptorrent/pen/VdogVg

Add data to chart seems to fix the bug but created a bad display for columns

**Comment by martynasma** [comment]
That's because your chart data is not empty. You have an object in there: `data: [{}]`.

Remove the `{}` ;)

**Comment by ptorrent** [comment]
Yes I add it for removing cursor bug :D

**Comment by martynasma** [comment]
Vicious circle :D

OK, so it definitely deserves some attention.

We'll need to investigate this properly, and keep you posted.

**Comment by martynasma** [comment]
Just an update, I've opened a separate issue for this: #158

Make sure you subscribe to it to follow progress.

**Comment by martynasma** [comment]
I'm closing this, but please feel free to post any follow up questions you may have.

Otherwise we'll update you with the progress of this fix in #158.
